tests/rpc-tests/main.go

```diff
@@ -0,0 +1,1969 @@
+		address, ok := log["address"].(string)
+		if !ok || !strings.EqualFold(address, stateSyncAddress) {
+			continue
+		}
+
+		topics, ok := log["topics"].([]interface{})
+		if !ok || len(topics) == 0 {
+			continue
+		}
+
+		firstTopic, ok := topics[0].(string)
+		if !ok {
+			continue
+		}
+
+		// Remove "0x" prefix if present
+		firstTopic = strings.TrimPrefix(firstTopic, "0x")
+		topicBytes, err := hex.DecodeString(firstTopic)
+		if err != nil {
+			continue
+		}
+
+		if bytes.Equal(topicBytes, expectedTopic) {
+			foundStateSyncLog = true
+			break
+		}
+	}
+
+	if !foundStateSyncLog {
+		return fmt.Errorf("state sync log not found")
+	}
+
+	return nil
+}
+
+func handleGetStateSyncBlockReceipts(rm *ResponseMap, method string, txReceipts *[]map[string]interface{}) error {
+
+	// state sync tx must always be the last one
+	stateSyncTx := (*txReceipts)[len(*txReceipts)-1]
+
+	err := validateStateSyncTxReceipt((*txReceipts)[len(*txReceipts)-1])
+	if err != nil {
+		return fmt.Errorf("last receipt on block must be state sync tx: %s", err)
+	}
+	txIndexHexStringWihtoutPrefix := (stateSyncTx)["transactionIndex"].(string)[2:]
+	txIndexOnMethod, err := strconv.ParseInt(txIndexHexStringWihtoutPrefix, 16, 0)
+	if err != nil {
+		return fmt.Errorf("error converting hex to int:%s", err)
+	}
+	if int(txIndexOnMethod) != rm.stateSyncTxIndex {
+		return fmt.Errorf("error on state sync tx index in block %s: transaction index on method %s: %d | transaction index on method eth_getTransactionReceipt: %d", rm.stateSyncBlockHash, method, txIndexOnMethod, rm.stateSyncTxIndex)
+	}
+
+	return nil
+}
+
+func validateHexBigInt(value interface{}, fieldName string, customValidation func(*big.Int) error) error {
+	strValue, ok := value.(string)
+	if !ok {
+		return fmt.Errorf("invalid type for %s, expected string", fieldName)
+	}
+
+	// Ensure it starts with "0x"
+	if len(strValue) < 3 || strValue[:2] != "0x" {
+		return fmt.Errorf("%s must be a hex string prefixed with 0x", fieldName)
+	}
+
+	// Convert to big.Int
+	bigValue, success := new(big.Int).SetString(strValue[2:], 16)
+	if !success {
+		return fmt.Errorf("invalid hex string for %s: %s", fieldName, strValue)
+	}
+
+	// Apply custom validation if provided
+	if customValidation != nil {
+		return customValidation(bigValue)
+	}
+
+	return nil
+}
+
+// ValidateCodeHash checks if the provided runtime code matches the account's codeHash
+func ValidateCodeHash(runtimeCode []byte, accountResult *accountResult) error {
+	calculatedCodeHash := crypto.Keccak256Hash(runtimeCode)
+
+	if calculatedCodeHash != accountResult.CodeHash {
+		return fmt.Errorf("code hash mismatch: calculated %s, account has %s",
+			calculatedCodeHash.Hex(), accountResult.CodeHash.Hex())
+	}
+	return nil
+}
+
+func generateInputForDeployTestContract(key string, value *big.Int) []byte {
+	abi, _ := testcontract.TestcontractMetaData.GetAbi()
+	input, _ := abi.Pack("", key, value)
+
+	return append(common.FromHex(testcontract.TestcontractBin), input...)
```

Comment:
`TestcontractBin` is deprecated, we should be using `TestcontractMetaData.Bin` instead.